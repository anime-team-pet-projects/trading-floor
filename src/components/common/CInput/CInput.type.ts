export type CInputType = 'textarea' | 'text' | 'color' | 'url' | 'search' | 'tel' | 'password' | 'email' | 'number'
