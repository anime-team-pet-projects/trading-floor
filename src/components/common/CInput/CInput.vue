<template>
  <el-input
    ref="inputInstance"
    :disabled="props.disabled"
    :model-value="props.modelValue"
    :placeholder="props.searchable ? props.label : ''"
    data-tid="c-input"
    v-bind="$attrs"
    v-on="$attrs"
    @update:model-value="handleInput"
  />
</template>

<script lang="ts" setup>
import { ElInput } from 'element-plus';
import { CInputType } from '@/components/common/CInput/CInput.type';

type Props = {
  modelValue: string;
  type?: CInputType;
  label?: string;
  disabled?: boolean;
  showPassword?: boolean;
  searchable?: boolean;
};

type Emits = {
  (e: 'update:modelValue', value: string): void;
  (e: 'focus'): void;
  (e: 'blur'): void;
};

const props = withDefaults(defineProps<Props>(), {
  modelValue: '',
  type: 'text',
  label: '',
  disabled: false,
  showPassword: false,
  searchable: false,
});
const emits = defineEmits<Emits>();

const handleInput = (value: string): void => {
  emits('update:modelValue', value);
};
</script>
