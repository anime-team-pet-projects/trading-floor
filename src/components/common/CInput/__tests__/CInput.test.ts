import { fireEvent, getByTestId, queryByTestId, configure } from '@testing-library/vue'
import { expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import CInput from '@/components/common/CInput/CInput.vue'

configure({ testIdAttribute: 'data-tid' })

//PO - PageObject
const CInputPO = {
  click() {
    fireEvent.click(getByTestId(document.body, 'c-input'))
  },
  getComponent() {
    return queryByTestId(document.body, 'c-input')
  },
}

test('CInput Snapshot', () => {
  const wrapper = mount(CInput, {
    props: { modelValue: 'snapshot' },
  })

  expect(wrapper.element).toMatchSnapshot()
})
