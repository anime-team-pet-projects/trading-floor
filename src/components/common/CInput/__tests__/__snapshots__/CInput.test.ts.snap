// Vitest Snapshot v1

exports[`CInput Snapshot 1`] = `
<div
  class="el-input"
>
  <!-- input -->
  
  <!-- prepend slot -->
  <!--v-if-->
  <div
    class="el-input__wrapper"
  >
    <!-- prefix slot -->
    <!--v-if-->
    <input
      autocomplete="off"
      class="el-input__inner"
      data-tid="c-input"
      placeholder=""
      tabindex="0"
      type="text"
    />
    <!-- suffix slot -->
    <!--v-if-->
  </div>
  <!-- append slot -->
  <!--v-if-->
  
</div>
`;
